Phong

justanavgme opened the pull request:
Added I2C library.

_Old version:_

> _I2C modules can now be selected using #define I2Cn_
> _Fast mode (400 Kbps) can be enabled/disabled using #define FAST_MODE_

_Current version:
I2C modules can now be selected
Fast mode (400 Kbps) can now be enabled/disabled._

**Both are done using 2 params I2C_SELECT and FAST_MODE in I2C_Config function.**
For example: `I2C_Config("I2C0", true)  //select module I2C0 & enable fast mode`

_Other minor changes:_
Added `#include <string.h>` library in I2C.h file to perform string comparison.
Changed parameter's name from `length` to `num` for representing the number of bytes to be sent/read.
Changed function name from `Write_bytes` and `Read_bytes` to `I2C_Write_bytes` and `I2C_Read_bytes`.

**All 4 I2C modules have been tested and worked fine when used to read MPU6050's data.**

Please comment "I approve" below. The pull request will be merged after all owners agreed.

justanavgme committed:
Merge pull request #4 from VGUHelloWorld/master

Remove comments

justanavgme committed:
Merge pull request #5 from VGUHelloWorld/master

README.md updated

justanavgme committed:
Update README.md

justanavgme committed:
Added I2C library

I2C modules can be chosen. Enable/disable FAST_MODE

justanavgme committed:
Merge branch 'Phong' of https://github.com/VGUHelloWorld/AutoBalanceBot into Phong

nkquynh98 commented on I2C/I2C.h:
```diff
@@ -0,0 +1,24 @@
+#define I2C1
+#define FAST_MODE false
+
+void I2C_Config(void);
```
Why don't you use the extern ?

justanavgme replied:
Thank you.
I've added "extern"(s) in the [6th commit](https://github.com/VGUHelloWorld/AutoBalanceBot/pull/6/commits/6d1fc8f482e91a9bed4614a0f82972fbca0f4f94).

nkquynh98 commented:
I approve

justanavgme committed:
Added extern

Added missing "extern"

justanavgme committed:
Updated I2C library

I2C module and enable/disable fast-mode can now be selected without editing the library's source code.

justanavgme committed:
Tested & built

All 4 I2C modules have been tested and worked fine when used to read MPU6050's data.

justanavgme committed:
Added comment

justanavgme commented:
@nkquynh98 @GiangNguyenHongLam 
Please check the I2C library's code again. I've made several major changes.
Details:

> _Current version:
> I2C modules can now be selected
> Fast mode (400 Kbps) can now be enabled/disabled._
> 
> **Both are done using 2 params I2C_SELECT and FAST_MODE in I2C_Config function.**
> For example: `I2C_Config("I2C0", true)  //select module I2C0 & enable fast mode`
> 
> _Other minor changes:_
> Added `#include <string.h>` library in I2C.h file to perform string comparison.
> Changed parameter's name from `length` to `num` for representing the number of bytes to be sent/read.
> Changed function name from `Write_bytes` and `Read_bytes` to `I2C_Write_bytes` and `I2C_Read_bytes`.

**Please let me know again if you approve this new version of the library**

nkquynh98 commented:
I approve

HongLamGiangNguyen commented:
I Approve
Sorry for such irresponsible act.
